src/notation/internal/notationinteraction.cpp

```diff
@@ -3495,3 +3650,55 @@ void NotationInteraction::toggleUnderline()
 {
     toggleFontStyle(Ms::FontStyle::Underline);
 }
+
+template<typename P>
+void NotationInteraction::execute(void (Ms::Score::* function)(P), P param)
+{
+    startEdit();
+    (score()->*function)(param);
+    apply();
+    notifyAboutNotationChanged();
+}
+
+void NotationInteraction::toggleArticulation(Ms::SymId symId)
+{
+    execute(&Ms::Score::toggleArticulation, symId);
+}
+
+void NotationInteraction::toggleAutoplace(bool all)
+{
+    execute(&Ms::Score::cmdToggleAutoplace, all);
+}
+
+void NotationInteraction::insertClef(Ms::ClefType clef)
+{
+    execute(&Ms::Score::cmdInsertClef, clef);
+}
+
+void NotationInteraction::changeAccidental(Ms::AccidentalType accidental)
+{
+    execute(&Ms::Score::changeAccidental, accidental);
```

Comment:
TODO: need a way of indicating that this is a pitch-change and requires playing back the currently selected note